pkg/jobs/registry.go

```diff
@@ -1202,7 +1203,7 @@ func (r *Registry) stepThroughStateMachine(
 ) error {
 	payload := job.Payload()
 	jobType := payload.Type()
-	log.Infof(ctx, "%s job %d: stepping through state %s with error: %+v", jobType, job.ID(), status, jobErr)
```

Comment:
We _could_ make it _slightly_ quieter by conditionally skipping it if the job is of type auto create stats, but i hesitate to even do that since we _have_ seen those stats jobs implicated in overload investigations or blocking server drain/shutdown which was  debugged via this line iirc.

Reply:
is there mileage to be gained with a different behavior when jobErr == nil?

Reply:
As dt mentions, in terms of debugging this is probably the most useful log message in this package.  Aside from just the error, it is kinda the only thing that lets you track where a job was running over time when looking at a long-running job that may have been adopted multiple times.

I'd prefer we keep this one at Info if we can as there is no way to get this information after the fact.

> We could make it slightly quieter by conditionally skipping it if the job is of type auto create stats

If we went this route, perhaps rather than switching on type, we have a resumer method that returns a bool that controls whether that job type gets verbose logging by default. Kinda like we have for ForceRealSpans.

Reply:
While failing with err is the most useful, knowing when it started and resumed are often very useful too, as is knowing when non-err jobs succeeded. This comes up if we're being asked to investigate "why did this backup take 6h instead of 3min" or "why did my all decommissioning block for 3 hours until 0437?"

The other log lines here happen at a constant rate (every time we look for expired jobs) or an unbounded number of times per job (which too and from idle), but this one here is mostly a fixed number of lines per job (eg create, run, failing, failed) and those are all pretty vital when doing forensics after the fact to figure out what was going on when

Reply:
I 100% agree that this is an incredibly useful log line.